simplemma/language_detector.py

```diff
@@ -10,24 +10,44 @@
 from .dictionary_factory import DictionaryFactory
 from .tokenizer import Tokenizer
 
+
 SPLIT_INPUT = re.compile(r"[^\W\d_]{3,}")
+RELAXED_SPLIT_INPUT = re.compile(r"[\w-]{3,}")
 
 
 class TokenSampler:
-    __slots__ = ["tokenizer"]
+    __slots__ = ["capitalized_threshold", "max_tokens", "tokenizer"]
 
-    def __init__(self, tokenizer: Tokenizer = Tokenizer(SPLIT_INPUT)) -> None:
+    def __init__(
+        self,
+        tokenizer: Tokenizer = Tokenizer(SPLIT_INPUT),
+        max_tokens: int = 1000,
```

Comment:
I'd make `max_tokens` optional. If none, I'd return the full list.

Likewise, I'd make `capitalized_threshold` optional and fi none, I'd removed all capitalized words.

Reply:
I'm more inclined to further reduce the value for max_tokens to 100. It's an important feature, default detection will be too slow on long texts otherwise.